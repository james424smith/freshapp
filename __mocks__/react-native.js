import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
