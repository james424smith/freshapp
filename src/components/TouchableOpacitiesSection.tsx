import React from "react";
import Text from "./StyledText";
import { TouchableOpacity, View } from "react-native";
import styles from "../styles/AssignmentScreenStyles";
import {
  AssignmentType,
  SeaServiceType,
  NewsType,
  SalaryType,
} from "../interfaces";
import { useTheme, Theme } from "@react-navigation/native";

type OptionsTypes = SeaServiceType | AssignmentType | NewsType | SalaryType;
type Props = {
  handleSelectedOption: (arg: unknown) => void;
  selectedOptionFromProps: OptionsTypes;
  firstOption: OptionsTypes;
  firstOptionLabel: string;
  secondOption: OptionsTypes;
  secondOptionLabel: string;
  backgroundColor?: string;
  fontColor?: string;
  fontWeight?: string;
};

function chooseTouchableOpacityStyle(
  menuOption: OptionsTypes,
  selectedOptionFromProps: OptionsTypes,
  fontColor?: string
) {
  return selectedOptionFromProps === menuOption
    ? [
        styles.activeTouchableOpacity,
        fontColor ? { borderBottomColor: fontColor } : {},
      ]
    : styles.inactiveTouchableOpacity;
}

function chooseTextStyle(
  theme: Theme,
  menuOption: OptionsTypes,
  selectedOptionFromProps: OptionsTypes,
  fontColor?: string,
  fontWeight?: string
) {
  return selectedOptionFromProps === menuOption
    ? [
        {
          ...styles.activeTouchableOpacityLabel,
          color: theme.colors.inactiveTouchableOpacityStyle,
        },
        fontColor ? { color: fontColor } : {},
        fontWeight ? { fontWeight: fontWeight } : {},
      ]
    : [
        {
          ...styles.inactiveTouchableOpacityLabel,
          color: theme.colors.inactiveTouchableOpacityStyle,
        },
      ];
}

function isDisabled(
  menuOption: OptionsTypes,
  selectedOptionFromProps: OptionsTypes
) {
  return menuOption === selectedOptionFromProps;
}

const TouchableOpacitiesSection = (props: Props) => {
  const {
    handleSelectedOption,
    selectedOptionFromProps,
    firstOptionLabel,
    secondOptionLabel,
    firstOption,
    secondOption,
    backgroundColor,
    fontColor,
    fontWeight,
  } = props;
  const theme = useTheme();
  return (
    <View
      style={[
        {
          ...styles.touchableOpacityContainer,
          backgroundColor: theme.colors.primary,
        },
        backgroundColor ? { backgroundColor: backgroundColor } : {},
      ]}
    >
      <TouchableOpacity
        testID="first-option-button"
        style={chooseTouchableOpacityStyle(
          firstOption,
          selectedOptionFromProps,
          fontColor
        )}
        onPress={() => handleSelectedOption(firstOption)}
        disabled={isDisabled(firstOption, selectedOptionFromProps)}
      >
        <Text
          style={chooseTextStyle(
            theme,
            firstOption,
            selectedOptionFromProps,
            fontColor,
            fontWeight
          )}
        >
          {firstOptionLabel}
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        testID="second-option-button"
        style={chooseTouchableOpacityStyle(
          secondOption,
          selectedOptionFromProps,
          fontColor
        )}
        onPress={() => handleSelectedOption(secondOption)}
        disabled={isDisabled(secondOption, selectedOptionFromProps)}
      >
        <Text
          style={chooseTextStyle(
            theme,
            secondOption,
            selectedOptionFromProps,
            fontColor,
            fontWeight
          )}
        >
          {secondOptionLabel}
        </Text>
      </TouchableOpacity>
    </View>
  );
};

export default TouchableOpacitiesSection;
