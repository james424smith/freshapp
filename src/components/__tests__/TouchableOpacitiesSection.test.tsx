import React from "react";
import TouchableOpacitiesSection from "../TouchableOpacitiesSection";
import { render, cleanup, fireEvent } from "@testing-library/react-native";

describe("TouchableOpacitiesSection snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui when everetyhing is definfed", () => {
    const handleSelectedOption = jest.fn();
    const { toJSON } = render(
      <TouchableOpacitiesSection
        handleSelectedOption={handleSelectedOption}
        selectedOptionFromProps={"Allotments"}
        firstOption={"Allotments"}
        firstOptionLabel={"allotments"}
        secondOption={"Payslips"}
        secondOptionLabel={"payslips"}
        backgroundColor="black"
        fontColor={"black"}
        fontWeight={"100"}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it("should render correct ui when all optional props are undefinfed", () => {
    const handleSelectedOption = jest.fn();
    const { toJSON } = render(
      <TouchableOpacitiesSection
        handleSelectedOption={handleSelectedOption}
        selectedOptionFromProps={"Allotments"}
        firstOption={"Allotments"}
        firstOptionLabel={"allotments"}
        secondOption={"Payslips"}
        secondOptionLabel={"payslips"}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

describe("TouchableOpacitiesSection firevent test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should call aprropriate action when first-option-button is press and option is not available", () => {
    const handleSelectedOption = jest.fn();
    const { getByTestId } = render(
      <TouchableOpacitiesSection
        handleSelectedOption={handleSelectedOption}
        selectedOptionFromProps={"Allotments"}
        firstOption={"Allotments"}
        firstOptionLabel={"allotments"}
        secondOption={"Payslips"}
        secondOptionLabel={"payslips"}
        backgroundColor="black"
        fontColor={"black"}
        fontWeight={"100"}
      />
    );
    fireEvent.press(getByTestId("first-option-button"));
    expect(handleSelectedOption).toHaveBeenCalledTimes(0);
  });

  it("should call aprropriate action when first-option-button is pressed and option is available", () => {
    const handleSelectedOption = jest.fn();
    const { getByTestId } = render(
      <TouchableOpacitiesSection
        handleSelectedOption={handleSelectedOption}
        selectedOptionFromProps={"Allotments"}
        firstOption={"Payslips"}
        firstOptionLabel={"payslips"}
        secondOption={"Allotments"}
        secondOptionLabel={"allotments"}
        backgroundColor="black"
        fontColor={"black"}
        fontWeight={"100"}
      />
    );
    fireEvent.press(getByTestId("first-option-button"));
    expect(handleSelectedOption).toHaveBeenCalledWith("Payslips");
  });
  it("should call aprropriate action when second-option-button is pressed and option is available", () => {
    const handleSelectedOption = jest.fn();
    const { getByTestId } = render(
      <TouchableOpacitiesSection
        handleSelectedOption={handleSelectedOption}
        selectedOptionFromProps={"Allotments"}
        firstOption={"Allotments"}
        firstOptionLabel={"allotments"}
        secondOption={"Payslips"}
        secondOptionLabel={"payslips"}
        backgroundColor="black"
        fontColor={"black"}
        fontWeight={"100"}
      />
    );
    fireEvent.press(getByTestId("second-option-button"));
    expect(handleSelectedOption).toHaveBeenCalledWith("Payslips");
  });
});
