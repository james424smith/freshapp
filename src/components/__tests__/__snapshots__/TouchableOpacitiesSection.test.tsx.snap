// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableOpacitiesSection snapshot test should render correct ui when all optional props are undefinfed 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "backgroundColor": "rgb(0, 122, 255)",
        "flexDirection": "row",
        "justifyContent": "space-around",
      },
      Object {},
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomColor": "#2A2E43",
        "borderBottomWidth": 2,
        "opacity": 1,
      }
    }
    testID="first-option-button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Array [
            Object {
              "color": undefined,
              "fontSize": 15,
              "fontWeight": "bold",
              "paddingBottom": 5,
            },
            Object {},
            Object {},
          ],
        ]
      }
    >
      allotments
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="second-option-button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Array [
            Object {
              "color": undefined,
              "fontSize": 14,
              "fontWeight": "bold",
              "opacity": 0.2,
            },
          ],
        ]
      }
    >
      payslips
    </Text>
  </View>
</View>
`;

exports[`TouchableOpacitiesSection snapshot test should render correct ui when everetyhing is definfed 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "backgroundColor": "rgb(0, 122, 255)",
        "flexDirection": "row",
        "justifyContent": "space-around",
      },
      Object {
        "backgroundColor": "black",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomColor": "black",
        "borderBottomWidth": 2,
        "opacity": 1,
      }
    }
    testID="first-option-button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Array [
            Object {
              "color": undefined,
              "fontSize": 15,
              "fontWeight": "bold",
              "paddingBottom": 5,
            },
            Object {
              "color": "black",
            },
            Object {
              "fontWeight": "100",
            },
          ],
        ]
      }
    >
      allotments
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="second-option-button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Array [
            Object {
              "color": undefined,
              "fontSize": 14,
              "fontWeight": "bold",
              "opacity": 0.2,
            },
          ],
        ]
      }
    >
      payslips
    </Text>
  </View>
</View>
`;
